docs/DeveloperGuide.md

```diff
@@ -1,38 +1,416 @@
+* The subclasses of `CommandParser` then serve to parse all arguments of a user input to create an object of a specific
+  subclass of the `Command` class.
+
+The general workflow of the `Parser` component is as follows:
+1. When required to parse for a command, the running `SplitLah` object passes a `String` object containing
+   the user input to `Parser` class.
+2. `Parser` class instantiates a new `XYZCommandParser` object corresponding to the user input 
+   and passes the user input to it.
+   (`XYZCommand` is a placeholder for specific subclass of the `Command` class, e.g. `SessionCreateCommand`)
+3. The `XYZCommandParser` object then uses parse methods from `Parser` class to extract all the
+   arguments from the user input.
+   * Each of these parse methods in `Parser` class then calls utility methods from `ParserUtils` class 
+     to return a parsed value.
+4. All relevant arguments that are parsed are then used to create a new `XYZCommand `object to be
+   returned to the `Parser` class.
+5. The created `XYZCommand` object is then returned to the `SplitLah` object to be run.
+
+### Command Component
+![Command Component Screenshot](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/CommandComponent.drawio.png)
+<br>
+The `Command` component consists of the Command class and all the subclasses of the Command class
+(also referred to as XYZCommand collectively).
+* The subclasses include _data related commands_ such as `SessionJKLCommand`, `ActivityJKLCommand` and `GroupJKLCommand`,
+  each representing commands that work with `Session`, `Activity` and `Group` classes respectively
+  (JKL here refers to either `Create`, `Delete`, `Edit`, `List`, `View` or `Summary`).
+* The subclasses also include _utility commands_ such as `HelpCommand`, `ExitCommand` and `InvalidCommand`
+  (hereafter collectively referred to as `UtilityCommand`).
+
+The general workflow of the `Command` component is as follows:
+1. After a `XYZCommand` object is created by the [`Parser` Component](#parser-component),
+   it is passed back to the `SplitLah` object to be run.
+   * In the above process, all necessary information for the execution of the command is passed to and
+     saved by the `XYZCommand` constructor.
+2. Then, `XYZCommand#run` is executed by `SplitLah`. `XYZCommand#run` carries out the task
+   that `XYZCommand` is designed to do.
+   * In general, for _data related commands_, `SessionJKLCommand`, `ActivityJKLCommand` and `GroupJKLCommand`
+     obtains the relevant `Session`, `Activity` and `Group` objects before operating upon them with relevant methods.
+   * On the other hand, for _utility commands_, `UtilityCommand` works with and uses methods from the `TextUI` component
+     to print messages and carry out their tasks.
+   * The inner workings of each of the `XYZCommand` classes can be seen in greater detail under the [Implementation section](#implementation).
```

Comment:
Good job on the formatting as well as clarification on certain ambiguous terms used, makes it very readable for the reader